### PR TITLE
sync from JHUGenMELA

### PR DESCRIPTION
**Pull request by hroskes**
sync from JHUGenMELA

### CODE REVIEW

File: MELA/fortran/mod_Parameters.F90 [hunk 1]
```diff
@@ -147,6 +147,8 @@ MODULE ModParameters
 real(8), public            :: M_Reso  = 125.0d0   *GeV      ! X resonance mass (spin 0, spin 1, spin 2)     (can be overwritten by command line argument)
 real(8), public            :: Ga_Reso = 0.00407d0 *GeV      ! X resonance width
 real(8), public            :: HiggsDecayLengthMM = 0d0      ! Higgs decay length in [mm]
+real(8), public            :: M_Reso2 = -1000.0d0 *GeV      ! second resonance mass (spin 0 in off-shell VBF)     (can be overwritten by command line argument)
+real(8), public            :: Ga_Reso2= 50.00     *GeV      ! second resonance width
```
Comment (usarica) [inline MELA/fortran/mod_Parameters.F90]: For these two lines, I would suggest just -1 and 0

File: MELA/fortran/mod_Parameters.F90 [hunk 2]
```diff
@@ -464,6 +467,190 @@ MODULE ModParameters
...
+   complex(8), public :: gh2zgs3  = (0.00d0,0d0)
+   complex(8), public :: gh2zgs4  = (0.00d0,0d0)
+   complex(8), public :: gh2gsgs2 = (0.00d0,0d0)
+   complex(8), public :: gh2gsgs3 = (0.00d0,0d0)
+   complex(8), public :: gh2gsgs4 = (0.00d0,0d0)
+
+!-- HVV contact terms
+!    logical, public :: UseVprime=.false.  ! false: contact interaction,  true: heavy Zprime propagator
+!    integer, public :: OnlyVVpr = 0       ! +1=only contact term for decay1, -1=only contact term for decay2, 0=all terms
+   complex(8), public :: eh2z_L_E  = (0.00d0,0d0)
```
Comment (usarica) [inline MELA/fortran/mod_Parameters.F90]: The second resonance contact terms should be removed. It is unclear how they will be implemented in MCFM, and their names are not consistent with first resonance.

File: MELA/fortran/mod_Parameters.F90 [hunk 3]
```diff
@@ -610,6 +797,7 @@ MODULE ModParameters
 integer, public, target :: NuM_ = 15
 integer, public, target :: NuT_ = 16
 integer, public, target :: Hig_ = 25
+integer, public, target :: Hig2_= 49
```
Comment (usarica) [inline MELA/fortran/mod_Parameters.F90]: No need for Hig2, Hig is enough. If second resonance couplings are set, better to transfer them to first resonance, or just give error message and quit the program.

Comment (usarica) [reply MELA/fortran/mod_Parameters.F90]: Also, "signal"=Hig+Hig2, and the only ways Hig or Hig2 are enabled are through gh or gh2 couplings and Mreso>=0 vs Mreso2>=0.

File: MELA/fortran/mod_Parameters.F90 [hunk 4]
```diff
@@ -1735,7 +1939,7 @@ FUNCTION IsABoson(PartType)
 implicit none
 logical :: IsABoson
 integer :: PartType
-  IsABoson = ( abs(PartType).eq.abs(Pho_) .or. abs(PartType).eq.abs(Z0_) .or. abs(PartType).eq.abs(Wp_) .or. abs(PartType).eq.abs(Hig_) )
+  IsABoson = ( abs(PartType).eq.abs(Pho_) .or. abs(PartType).eq.abs(Z0_) .or. abs(PartType).eq.abs(Wp_) .or. abs(PartType).eq.abs(Hig_) .or. abs(PartType).eq.abs(Hig2_) )
```
Comment (usarica) [inline MELA/fortran/mod_Parameters.F90]: Same for all these lines, remove Hig2

**Comment by hroskes** [comment]
This is all from Markus.  I was just syncing with JHUGen.  Talk to him if you like, or remove it from JHUGen and then we'll sync back here.

I don't care either way about the second resonance, we can remove it or not as far as I'm concerned, but we should be consistent between JHUGen, JHUGenMELA, and CMSMELA.  It would be a really big mistake to diverge now.  It's hard enough as it is.

**Comment by usarica** [comment]
Ok, strange... I will remove them tomorrow, and hoefully we can finally close the sync. There is also something about a new success argument, do we need it somehow? (or rather is it related to Hig2?)

**Comment by hroskes** [comment]
We need it in JHUGen, that was me just now.  The logic for M_Wprime and Ga_Wprime gets complicated, I check several things.

**Comment by usarica** [comment]
Ah ok, so I won't touch it.

**Comment by usarica** [comment]
MELA/interface/RooSpinTwo_7D.h unnecessary, remove this file

**Pull request merged**